src/hypothesis/core.py

```diff
@@ -192,32 +191,36 @@ def wrapped_test(*arguments, **kwargs):
                 'destructuring arguments') % (
                     name,
             ))
-    if original_argspec.defaults:
-        return invalid(
-            'Cannot apply @given to a function with defaults.'
-        )
+    if original_argspec.defaults or original_argspec.kwonlydefaults:
+        return invalid('Cannot apply @given to a function with defaults.')
+    missing = [repr(kw) for kw in original_argspec.kwonlyargs
+               if kw not in generator_kwargs]
+    if missing:
+        raise InvalidArgument('Missing required kwarg{}: {}'.format(
+            's' if len(missing) > 1 else '', ', '.join(missing)))
 
 
 def execute_explicit_examples(
     test_runner, test, wrapped_test, settings, arguments, kwargs
 ):
-    original_argspec = getargspec(test)
+    original_argspec = getfullargspec(test)
 
     for example in reversed(getattr(
         wrapped_test, 'hypothesis_explicit_examples', ()
     )):
+        example_kwargs = dict(original_argspec.kwonlydefaults or {})
```

Comment:
I honestly don't know.  Either way, this is locally and more obviously correct, so I'll leave it as-is.